### PR TITLE
4656 filetable paginator

### PR DESCRIPTION
**Pull request by pdurbin**
connects to #4656

### CODE REVIEW

File: src/main/java/edu/harvard/iq/dataverse/DatasetPage.java [hunk 1]
```diff
@@ -4283,4 +4296,29 @@ public String getJsonLd() {
         }
         return "";
     }
+
+    public void selectAllFiles() {
+        logger.info("selectAllFiles called");
+        List<FileMetadata> allFiles = new ArrayList<>();
+        for (FileMetadata fm : workingVersion.getFileMetadatas()) {
```
Comment (oscardssmith) [inline src/main/java/edu/harvard/iq/dataverse/DatasetPage.java]: why not just `selectedFiles = workingVersion.getFileMetadatas();`

File: src/main/java/edu/harvard/iq/dataverse/DatasetPage.java [hunk 2]
```diff
@@ -4283,4 +4296,29 @@ public String getJsonLd() {
...
+        selectedFiles = Collections.EMPTY_LIST;
+    }
+    
+    public void fileListingPaginatorListener(PageEvent event) {
+        setFilePaginatorPage(event.getPage());      
+    }
+    
+    public void refreshPaginator() {
+        FacesContext facesContext = FacesContext.getCurrentInstance();
+        org.primefaces.component.datatable.DataTable dt = (org.primefaces.component.datatable.DataTable) facesContext.getViewRoot().findComponent("datasetForm:tabView:filesTable");
```
Comment (oscardssmith) [inline src/main/java/edu/harvard/iq/dataverse/DatasetPage.java]: importing `DataTable` would make this line much clearer. ie. `DataTable dt = (DataTable) facesContext.getViewRoot().findComponent("datasetForm:tabView:filesTable");`

Comment (sekmiller) [reply src/main/java/edu/harvard/iq/dataverse/DatasetPage.java]: The problem here is that there's already a DataTable object defined in dataverse so the ugliness is in line is related to making the distinction with the native DataTable object. If there's a cleaner way to make the distinction...

Comment (oscardssmith) [reply src/main/java/edu/harvard/iq/dataverse/DatasetPage.java]: Oh yeah, that sucks. You could import up to `component`. It's not ideal, but I think better.

Comment (oscardssmith) [reply src/main/java/edu/harvard/iq/dataverse/DatasetPage.java]: how I wish for python's `import x as y`

Comment (pdurbin) [reply src/main/java/edu/harvard/iq/dataverse/DatasetPage.java]: This isn't worth changing.

File: src/main/java/edu/harvard/iq/dataverse/DatasetPage.java (hunk 1 above)
Comment (pdurbin) [reply src/main/java/edu/harvard/iq/dataverse/DatasetPage.java]: @oscardssmith good idea. Fixed in c592236.

**Pull request merged**